src/components/checkbox/index.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { forwardRef, ReactNode } from 'react';
+import {
+  BoxProps,
+  Checkbox as ChakraCheckbox,
+  RecipeVariantProps,
+  SwitchCheckedChangeDetails,
+  useSlotRecipe,
+} from '@chakra-ui/react';
+
+import { checkboxRecipe } from 'src/themes/shared/slotRecipes/checkbox';
+
+type CheckboxVariantProps = RecipeVariantProps<typeof checkboxRecipe>;
+
+export type CheckboxProps = CheckboxVariantProps & {
+  name: string;
+  value?: string;
+  disabled?: boolean;
+  checked?: boolean;
+  invalid?: boolean;
+  indeterminate?: boolean;
+  readOnly?: boolean;
+  fullWidth?: boolean;
+  label?: ReactNode | string;
+  paddingY?: BoxProps['paddingY'];
+  fontWeight?: 'regular' | 'bold';
+  variant?: 'alignCenter' | 'alignTop' | 'alignTopForSmallSize';
+  onChange?: (details: SwitchCheckedChangeDetails) => void;
+};
+
+export const Checkbox = forwardRef<HTMLInputElement, CheckboxProps>(
+  (
+    {
+      name,
+      value,
+      disabled = false,
+      checked = false,
+      invalid,
+      indeterminate = false,
+      readOnly = false,
+      fullWidth = false,
+      label,
+      paddingY,
+      fontWeight = 'regular',
+      variant = 'alignCenter',
+      onChange,
+      ...props
+    },
+    ref,
+  ) => {
+    const recipe = useSlotRecipe({ key: 'checkbox' });
+    const combinedProps = { ...props, variant };
+    const [recipeProps] = recipe.splitVariantProps(combinedProps);
+    const styles = recipe(recipeProps);
+
+    return (
+      <ChakraCheckbox.Root
+        {...props}
+        name={name}
```

Comment:
I think you can drop everything where we don't set a default and the `...props` will pass it